Client/WebFilters/AdblockResourceDownloader.swift

```diff
@@ -46,49 +46,54 @@ class AdblockResourceDownloader {
+                do {
+                    try await withThrowingTaskGroup(of: Void.self) { [weak self] group in
+                        guard let self = self else { return }
+                        group.addTask { try await self.regionalAdblockResourcesSetup() }
+                        group.addTask { try await self.generalAdblockResourcesSetup() }
+                        try await group.waitForAll()
+                    }
+                } catch {
+                    log.error("Failed to Download Adblock-Resources: \(error)")
+                }
+            }
         }
     }
     
-    func regionalAdblockResourcesSetup() {
+    func regionalAdblockResourcesSetup() async throws {
         if !Preferences.Shields.useRegionAdBlock.value {
             log.debug("Regional adblocking disabled, aborting attempt to download regional resources")
             return
         }
         
-        downloadResources(type: .regional(locale: locale),
-                          queueName: "Regional adblock setup").uponQueue(.main) {
-                            log.debug("Regional blocklists download and setup completed.")
-                            Preferences.Debug.lastRegionalAdblockUpdate.value = Date()
-        }
+        try await downloadResources(type: .regional(locale: locale),
+                                    queueName: "Regional adblock setup")
+        log.debug("Regional blocklists download and setup completed.")
+        Preferences.Debug.lastRegionalAdblockUpdate.value = Date()
     }
     
-    func generalAdblockResourcesSetup() {
-        downloadResources(type: .general,
-                          queueName: "General adblock setup").uponQueue(.main) {
-                            log.debug("General blocklists download and setup completed.")
-                            Preferences.Debug.lastGeneralAdblockUpdate.value = Date()
-        }
+    func generalAdblockResourcesSetup() async throws {
+        try await downloadResources(type: .general,
+                                    queueName: "General adblock setup")
+        log.debug("General blocklists download and setup completed.")
+        Preferences.Debug.lastGeneralAdblockUpdate.value = Date()
     }
     
-    private func downloadResources(type: AdblockerType, queueName: String) -> Deferred<()> {
-        let completion = Deferred<()>()
-
-        let queue = DispatchQueue(label: queueName)
+    private func downloadResources(type: AdblockerType, queueName: String) async throws {
         let nm = networkManager
         let folderName = AdblockResourceDownloader.folderName
         
         // file name of which the file will be saved on disk
         let fileName = type.identifier
         
-        let completedDownloads = type.associatedFiles.map { fileType -> Deferred<AdBlockNetworkResource> in
+        async let completedDownloads = type.associatedFiles.asyncCompactMap { fileType -> AdBlockNetworkResource? in
```

Comment:
This is sequential despite it being `async let` because you're using `asyncCompactMap` which await's before running on the next element instead of using AsyncSequence (`for x in await y`) which will run each task concurrently and execute each time one finishes.  Chances are if you want each to run concurrently you'll want a `TaskGroup`